subprojects/parseq/src/main/java/com/linkedin/parseq/BaseTask.java

```diff
@@ -248,7 +248,11 @@ public String getName() {
   public boolean cancel(final Exception rootReason) {
     if (transitionCancel(rootReason)) {
       final Exception reason = new CancellationException(rootReason);
-      traceFailure(reason);
+      try {
+        traceFailure(reason);
+      } catch (Throwable ex) {
+        LOGGER.warn("Exception thrown in logging trace for failure!", ex);
```

Comment:
The issue we found was when the original `reason` could not be logged due to an error in its `toString`. Do we have any concern that logging the _new_ error might have the same issue? Would it be better to have two stages to this?

1. Catch the logging error, then try to log that error.
2. Catch _that_ error if it happens, and log a generic, static message.